lambdas/status_reports/src/t4_lambda_status_reports/__init__.py

```diff
@@ -0,0 +1,291 @@
+                    <td>
+                        {% if canary.lastRun %}
+                            {{ canary.lastRun }}
+                        {% else %}
+                            N/A
+                        {% endif %}
+                    </td>
+
+                </tr>
+            {% endfor %}
+        </tbody>
+    </table>
+
+    <h2>Installation Qualification</h2>
+
+    <h3>Stack Resources</h3>
+    <table>
+        <thead>
+            <tr>
+                <th>Logical ID</th>
+                <th>Physical ID</th>
+                <th>Type</th>
+                <th>Status</th>
+                <th>Last Updated</th>
+            </tr>
+        </thead>
+        <tbody>
+            {% for r in resources %}
+                <tr>
+                    <td>{{ r.LogicalResourceId }}</td>
+                    <td>{{ r.PhysicalResourceId }}</td>
+                    <td>{{ r.ResourceType }}</td>
+                    <td>{{ r.ResourceStatus }}</td>
+                    <td>{{ r.LastUpdatedTimestamp.isoformat() }}</td>
+                </tr>
+            {% endfor %}
+        </tbody>
+    </table>
+
+    <h3>Stack Outputs</h3>
+    <table>
+        <thead>
+            <tr>
+                <th>Output Key</th>
+                <th>Value</th>
+                <th>Description</th>
+            </tr>
+        </thead>
+        <tbody>
+            {% for o in stack_data.Outputs %}
+                <tr>
+                    <td>{{ o.OutputKey }}</td>
+                    <td>{{ o.OutputValue }}</td>
+                    <td>{{ o.Description }}</td>
+                </tr>
+            {% endfor %}
+        </tbody>
+    </table>
+
+    <h3>Stack Parameters</h3>
+    <table>
+        <thead>
+            <tr>
+                <th>Parameter Key</th>
+                <th>Value</th>
+            </tr>
+        </thead>
+        <tbody>
+            {% for p in stack_data.Parameters %}
+                <tr>
+                    <td>{{ p.ParameterKey }}</td>
+                    <td>{{ p.ParameterValue }}</td>
+                </tr>
+            {% endfor %}
+        </tbody>
+    </table>
+</body>
+</html>
+""")
+
+
+async def generate_status_report(stack_name: str):
+    async with create_cfn() as cfn, create_syn() as syn:
+        now = datetime.datetime.utcnow()
+        canaries, resources, stack_data = await asyncio.gather(
+            get_canaries(syn, cfn, stack_name),
+            get_resources(cfn, stack_name),
+            get_stack_data(cfn, stack_name),
+        )
+        html = tmpl.render(
+            stack_name=stack_name,
+            aws_region=AWS_REGION,
+            now=now,
+            canaries=canaries,
+            resources=resources,
+            stack_data=stack_data,
+        )
+        return now, html
+
+
+def async_handler(f):
+    @functools.wraps(f)
+    def wrapper(event, context):
+        return asyncio.run(f(event, context))
+
+    return wrapper
+
+
+@async_handler
+async def lambda_handler(*_):
```

Comment:
I'd use this for uniformity, but that's up to you.

```suggestion
async def lambda_handler(event, context):
```

Reply:
but these args are unused, so my local linter(ish) barfs